mPowerSDK/Dashboard/APHLineGraphView.m

```diff
@@ -70,4 +103,138 @@ - (void)drawXAxis
     }
 }
 
+- (void)drawGraphForPlotIndex:(NSInteger)plotIndex {
+	[super drawGraphForPlotIndex:plotIndex];
+
+	if (self.shouldDrawLastPoint) {
+		[self drawLastPointDot:plotIndex];
+	}
+}
+
+- (void)drawLastPointDot:(NSInteger)plotIndex {
+	__block NSInteger i = -1;
+	[self.dataPoints enumerateObjectsUsingBlock:^(NSNumber *dataPoint, NSUInteger idx, BOOL *stop) {
+		if (dataPoint.floatValue != NSNotFound) {
+			i = idx;
+		}
+	}];
+	
+	if (i < 0) return;
+	
+	CGFloat positionOnXAxis = ((NSNumber *)self.xAxisPoints[i]).floatValue;
+	CGFloat positionOnYAxis = ((NSNumber *)self.yAxisPoints[i]).floatValue;
+
+	CGFloat pointSize = 6.0f;
+	APCCircleView *point = [[APCCircleView alloc] initWithFrame:CGRectMake(0, 0, pointSize, pointSize)];
+	point.tintColor = (plotIndex == 0) ? self.colorForFirstCorrelationLine : self.colorForSecondCorrelationLine;
+	point.shapeLayer.fillColor = point.tintColor.CGColor;
+	point.center = CGPointMake(positionOnXAxis, positionOnYAxis);
+	[self.plotsView.layer addSublayer:point.layer];
+
+	if (self.shouldAnimate) {
+		point.alpha = 0;
+	}
+
+	[self.dots addObject:point];
+}
+
+- (void)drawLinesForPlotIndex:(NSInteger)plotIndex
+{
+    UIBezierPath *fillPath = [UIBezierPath bezierPath];
+    
+    CGPoint position = CGPointMake(CGFLOAT_MAX, CGFLOAT_MAX);
+    CGPoint prevPosition = CGPointMake(CGFLOAT_MAX, CGFLOAT_MAX);
+    
+    NSMutableArray *pointsArray = [NSMutableArray new];
+    
+    NSUInteger smallestArrayCount = self.yAxisPoints.count < self.xAxisPoints.count ?: self.xAxisPoints.count;
```

Comment:
I do not think that this is doing what you want it to be doing:

if `self.yAxisPoints.count=4` and `self.xAxisPoints.count=5` then `smallestArrayCount=1` or `true` in c syntax.